lib/rubocop/target_ruby.rb

```diff
@@ -192,15 +188,27 @@ def version_from_gemspec_file(file)
         required_ruby_version(processed_source.ast).first
       end
 
+      def version_from_right_hand_side(right_hand_side)
+        if right_hand_side.array_type?
+          version_from_array(right_hand_side)
+        elsif gem_requirement?(right_hand_side)
+          right_hand_side.children.last.value
+        else
+          right_hand_side.value
+        end
+      end
+
       def version_from_array(array)
-        versions = array.children.map { |v| version_from_str(v.is_a?(String) ? v : v.str_content) }
-        versions.compact.min
+        array.children.map(&:value)
       end
 
-      def version_from_str(str)
-        str.match(/^(?:>=|<=)?\s*(?<version>\d+(?:\.\d+)*)/) do |md|
-          md[:version].to_f
-        end
+      def find_minimal_known_ruby(right_hand_side)
+        version = version_from_right_hand_side(right_hand_side)
+        requirement = Gem::Requirement.new(version)
+        gem_version = ->(ruby_version) { Gem::Version.new(ruby_version) }
+
+        match = KNOWN_RUBIES.each.lazy.map(&gem_version).detect { |v| requirement.satisfied_by?(v) }
```

Comment:
Can't these lines be replaced by:
```
match = KNOWN_RUBIES.detect { |v| requirement.satisfied_by?(Gem::Version.new(v) }
```

I'd actually use `Gem::Version.new("#{v}.99")` so that it works ok for `>= '2.6.1'`

In general, storing a lambda in a local variable is an anti-pattern. Also, it's typically best to use `lazy` only when actually needed.

Reply:
Only one question. The `.99` would cause a not equal check like the ones I have [here](https://github.com/rubocop-hq/rubocop/pull/9515/files#diff-9ac8fa18e6383d5acc2f64f888efd29cd782ee80a85fa2c6e3090421f0d3e228R344) and [here](https://github.com/rubocop-hq/rubocop/pull/9515/files#diff-9ac8fa18e6383d5acc2f64f888efd29cd782ee80a85fa2c6e3090421f0d3e228R410) to fail. Maybe that is ok? Not sure how common it is to be specifying not equal versions

Reply:
Seems to me it is an error to use `!=`. In any case, if someone did specifies `!= 2.4.0`, that doesn't exclude `2.4.1` etc which are in the `2.4` series, so it seems like an ok target.

Reply:
Ok 1 more thing if we had `'> 2.4.1'` then it would match to `2.4` due to the `.99`, that seems wrong?

Reply:
I think that's the best behavior. Imagine that there is a bug in 2.4.1 that is fixed in Ruby 2.4.2+; the gem author wants to make sure we don't try to run the gem in 2.4.1 or below as the bug can be triggered by code in the gem. The gem still uses the "2.4.x" features and syntax, so target of 2.4 is best.

Reply:
Ok push the updates :)